bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/api/conditions/ConditionTransportRegistry.java

```diff
@@ -14,14 +14,53 @@
 
 import java.util.Map;
 
+/**
+ * Contract for a registry of {@link ConditionTransport}s.
+ *
+ * <p>As {@link LicensingCondition} can be persisted in any form, a {@link ConditionTransport}
+ * is associated with a particular  <i>content type</i>.
+ * </p>
+ *
+ * @see ConditionTransport
+ * @see LicensingCondition
+ * @see #getConditionTransportForContentType(String) 
+ * @since 0.4.0
+ */
 public interface ConditionTransportRegistry {
 
+    /**
+     * Returns an aggregate of all registered {@link ConditionTransport}s.
+     *
+     * @return all registered {@link ConditionTransport}s.
+     */
 	Iterable<ConditionTransport> getConditionTransports();
 
+    /**
+     * <p>
+     * Returns {@link ConditionTransport}, which is registered to handle (read and write)
+     * {@link LicensingCondition}s in the given {@code contentType}. The value is nullable.
+     * </p>
+     *
+     * @param contentType string representation of content type (like <i>application/json</i> or <i>application/xmk</i>)
+     * @return a {@link ConditionTransport} registered for the given {@code contentType}, if any, and {@code null} otherwise.
+     * @see #registerConditionTransport(ConditionTransport, Map)
+     */
 	ConditionTransport getConditionTransportForContentType(String contentType);
```

Comment:
Now I see that <Optional> will be much better here, No way to change it now, but we can add another "find" method and deprecate this one.